tests/cypress/page-object/users.page.ts

```diff
@@ -0,0 +1,27 @@
+import { BasePage } from './base.page'
+
+class UsersPage extends BasePage {
```

Comment:
Wondering is this cypress standard to have action in classes?

Reply:
yeah, this is the page object approach.
The Cypress team themselves recommend a different approach, but most implementations still use the page object approach

Reply:
May you kindly elaborate why we decided to use the page object approach?

Reply:
there are two approaches on how to structure the test code, Page Object and App Actions.
The logic for App Actions is presented by the Cypress team in this [blog post ](https://www.cypress.io/blog/2019/01/03/stop-using-page-objects-and-start-using-app-actions/).
Most implementation are still choosing the Page Object approach because in the end it gives a more structured code and less code duplication.
Example reinforcing those benefits: [blog](https://automationrhapsody.com/testing-with-cypress-lessons-learned-in-a-complete-framework/)
I actually couldn't find at the time a large project using the App Action approach.
Just FYI, we're still using App Actions, just not as a replacement for the Page Object model.